source/common/common/utility.cc

```diff
@@ -98,6 +98,27 @@ void StringUtil::rtrim(std::string& source) {
   }
 }
 
+void StringUtil::ltrim(std::string& source) {
+  std::size_t pos = source.find_first_not_of(" \t\f\v\n\r");
```

Comment:
If you use absl::string_view you will not have to copy/allocate any string data for this operation.

https://github.com/abseil/abseil-cpp/blob/master/absl/strings/string_view.h

absl is already incorporated into the Envoy build.


I admit that you may need to construct a temp std::string later to do the hash lookup though, but in one case it would at least save you one intermediate copy.  You could also avoid that intermediate copy by using std::unordered_set<absl::string_view> but you'll need to keep the backing store for the strings separately then.  Not sure if that's worth it, but it would save allocations/string-copying per request in the server.